floris/simulation/wake_velocity/turbopark.py

```diff
@@ -0,0 +1,262 @@
+
+                -   **we** (*float*): The linear wake decay constant that
+                    defines the cone boundary for the wake as well as the
+                    velocity deficit. D/2 +/- we*x is the cone boundary for the
+                    wake.
+        """
+        super().__init__(parameter_dictionary)
+        self.model_string = "jensen"
+        model_dictionary = self._get_model_dict(__class__.default_parameters)
+        self.A = float(model_dictionary["A"])
+        self.c1 = float(model_dictionary["c1"])
+        self.c2 = float(model_dictionary["c2"])
+
+    def function(
+        self,
+        x_locations,
+        y_locations,
+        z_locations,
+        turbine,
+        turbine_coord,
+        deflection_field,
+        flow_field,
+    ):
+        """
+        Using the Jensen wake model, this method calculates and returns
+        the wake velocity deficits, caused by the specified turbine,
+        relative to the freestream velocities at the grid of points
+        comprising the wind farm flow field.
+
+        Args:
+            x_locations (np.array): An array of floats that contains the
+                streamwise direction grid coordinates of the flow field
+                domain (m).
+            y_locations (np.array): An array of floats that contains the grid
+                coordinates of the flow field domain in the direction normal to
+                x and parallel to the ground (m).
+            z_locations (np.array): An array of floats that contains the grid
+                coordinates of the flow field domain in the vertical
+                direction (m).
+            turbine (:py:obj:`floris.simulation.turbine`): Object that
+                represents the turbine creating the wake.
+            turbine_coord (:py:obj:`floris.utilities.Vec3`): Object containing
+                the coordinate of the turbine creating the wake (m).
+            deflection_field (np.array): An array of floats that contains the
+                amount of wake deflection in meters in the y direction at each
+                grid point of the flow field.
+            flow_"Updatefield (:py:class:`floris.simulation.flow_field`): Object
+                containing the flow field information for the wind farm.
+
+        Returns:
+            np.array, np.array, np.array:
+                Three arrays of floats that contain the wake velocity
+                deficit in m/s created by the turbine relative to the freestream
+                velocities for the U, V, and W components, aligned with the
+                x, y, and z directions, respectively. The three arrays contain
+                the velocity deficits at each grid point in the flow field.
+        """
+
+        # Get model parameters
+        A = self.A
+        c1 = self.c1
+        c2 = self.c2
+
+        # TODO Not positive right way to grab this value:
+        # Is the right one flow_field.u_initial?
```

Comment:
Looking at this line: https://github.com/NREL/floris/blob/main/floris/simulation/flow_field.py#L238
I think `wind_map.grid_wind_speed` is what you want.

Reply:
Would that require new input to the class?